e2e/cypress/integration/messaging/permalink_loading_indicator_spec.js

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Permalink loading indicator', () => {
+    before(() => {
+        // # Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M18701-Permalink to first post in channel shows endless loading indicator above', () => {
+        const dateNow = Date.now();
+        const message = 'Hello' + dateNow;
+        const privateChannelName = 'test-private-channel-' + dateNow;
+        const publicChannelName = 'test-public-channel-' + dateNow;
+
+        // # Get current team id
+        cy.getCurrentTeamId().then((teamId) => {
+            // # Create private channel to post message
+            cy.apiCreateChannel(teamId, privateChannelName, privateChannelName, 'P', 'Test private channel').then((response) => {
+                const testPrivateChannel = response.body;
+
+                // # Click on test private channel
+                cy.get('#sidebarItem_' + testPrivateChannel.name).click({force: true});
+
+                // # Post message to use
+                cy.postMessage(message);
```

Comment:
On step 2, should post few more messages (could be 10 more).

Reply:
Updated in latest commit.